keyboards/bastardkb/charybdis/charybdis.c

```diff
@@ -0,0 +1,245 @@
+/* Copyright 2020 Christopher Courtney <drashna@live.com> (@drashna)
+ * Copyright 2021 Quentin LEBASTARD <qlebastard@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Publicw License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "charybdis.h"
+#include "transactions.h"
+#include <string.h>
+#include "drivers/sensors/pmw3360.h"
+#include "print.h"
+
+#ifndef CHARYBDIS_DPI_OPTIONS
+#    define CHARYBDIS_DPI_OPTIONS \
+        { 600, 800, 1000, 1200, 1400, 1600, 1800, 2000, 2200, 2400, 2600, 2800  }
+#endif
+#ifndef CHARYBDIS_DPI_DEFAULT
+#    define CHARYBDIS_DPI_DEFAULT 0
+#endif
+
+#ifndef CHARYBDIS_SNIPER_OPTIONS
+#define CHARYBDIS_SNIPER_OPTIONS \
+        { 100, 200, 300, 400 }
+#endif
+#ifndef CHARYBDIS_SNIPER_DEFAULT
+#    define CHARYBDIS_SNIPER_DEFAULT 0
+#endif
+#ifndef CHARYBDIS_AUTO_SNIPER_LAYER
+#    define CHARYBDIS_AUTO_SNIPER_LAYER 4
+#endif
+
+#ifndef CHARYBDIS_DRAGSCROLL_DPI
+#    define CHARYBDIS_DRAGSCROLL_DPI 100 // Fixed-DPI Drag Scroll
+#endif
+
+#ifndef CHARYBDIS_DRAGSCROLL_MULTIPLIER
+#    define CHARYBDIS_DRAGSCROLL_MULTIPLIER 0.2 // Variable-DPI Drag Scroll
+#endif
+
+
+#ifndef CHARYBDIS_TRACKBALL_SPEED_DIVIDER
+#    define CHARYBDIS_TRACKBALL_SPEED_DIVIDER 12 
+#endif
+
+keyboard_config_t keyboard_config;
+uint16_t          dpi_array[] = CHARYBDIS_DPI_OPTIONS;
+#define DPI_OPTION_SIZE (sizeof(dpi_array) / sizeof(uint16_t))
+
+uint16_t          sniper_array[] = CHARYBDIS_SNIPER_OPTIONS;
+#define SNIPER_OPTION_SIZE (sizeof(sniper_array) / sizeof(uint16_t))
+
+bool    is_drag_scroll    = false;
+
+scroll_inertia_t scroll_inertia; // Scroll value storage to make scrolling slower
+
+__attribute__((weak)) void process_mouse_user(report_mouse_t* mouse_report, int16_t x, int16_t y) {
+    mouse_report->x = x;
+    mouse_report->y = y;
+}
+
+__attribute__((weak)) void process_mouse(report_mouse_t* mouse_report) {
+    report_pmw_t data = pmw_read_burst();
+    if (data.isOnSurface && data.isMotion) {
+        // Wrap to HID size
+        data.dx = constrain(data.dx, -127, 127);
+        data.dy = constrain(data.dy, -127, 127);
+
+        mouse_report->x = -data.dx;
+        mouse_report->y = data.dy;
```

Comment:
Otherwise, `process_mouse_user` isn't getting called. 
```suggestion
        process_mouse_user(mouse_report, -data.dx, data.dy);
```